components/telemetry-operator/controller/tracepipeline/controller.go

```diff
@@ -117,7 +117,8 @@ func (r *Reconciler) installOrUpgradeOtelCollector(ctx context.Context, tracing
 		return fmt.Errorf("failed to create otel collector configmap: %w", err)
 	}
 
-	deployment := makeDeployment(r.config)
+	configHash := NewConfigHash().AddByteMap(secretData).AddStringMap(configMap.Data).Build()
```

Comment:
Since order of `AddByteMap` and `AddStringMap` matters and we want to also make this internal of a function? 
Also with a signature like `CreateConfigHash(secrets []v1.Secret, cms []configsMaps)`